pkg/auth/serviceaccount.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/ptr"
+)
+
+// GetServiceAccountName returns the service account name to use for OIDC
+// authentication for the given resource.
+func GetServiceAccountName(gvk schema.GroupVersionKind, objectMeta metav1.ObjectMeta) string {
```

Comment:
GetServiceAccountName  For Resource?

Reply:
Do you mean to rename the function to `GetServiceAccountNameForResource` or is it unclear what I mean with the function description?

Reply:
@matzew ^

Reply:
yeah, the name `GetServiceAccountName` seem pretty generic

Reply:
Updated it